src/Umbraco.Tests.UnitTests/Umbraco.Web.Common/Formatters/IgnoreRequiredAttributesResolverUnitTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.ComponentModel.DataAnnotations;
+using System.Runtime.Serialization;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using Umbraco.Web.Common.Formatters;
+
+namespace Umbraco.Tests.UnitTests.Umbraco.Web.Common.Formatters
+{
+    [TestFixture]
+    public class IgnoreRequiredAttributesResolverUnitTests
+    {
+
+        [Test]
+        public void Test()
+        {
+            const string emptyJsonObject = "{}";
+
+            Assert.Multiple(() =>
+            {
+                //Ensure the deserialization throws if using default settings
+                Assert.Throws<JsonSerializationException>(() =>
+                    JsonConvert.DeserializeObject<ObjectWithRequiredProperty>(emptyJsonObject));
+
+                var actual = JsonConvert.DeserializeObject<ObjectWithRequiredProperty>(emptyJsonObject,new JsonSerializerSettings
+                {
+                    ContractResolver = new IgnoreRequiredAttributesResolver()
+                });
+
+                Assert.IsNotNull(actual);
+                Assert.IsNull(actual.Property);
+            });
+        }
+
+        [DataContract(Name = "objectWithRequiredProperty", Namespace = "")]
+        private class ObjectWithRequiredProperty
+        {
+            [DataMember(Name = "property", IsRequired = true)]
```

Comment:
So am I correct in assuming that this new IgnoreRequiredAttributesResolver is specifically for deserializing JSON so it actually deserializes without an exception even when IsRequired is there *but* the IsRequired will still be used during validation? Or is the validation Required attribute different? If so, is there ever a need to use the IsRequired param?

Reply:
Yes..
The `RequiredAttribute` is different. I guess that is just used invalidation. 

Maybe we could just remove the `DataMember(IsRequired = true)`. I assumed it was there because we require it hen we output the same model.

Reply:
maybe, i think we could revisit it, maybe for now we'll just leave this as-is. seems strange that it worked in v8, or maybe that's only because we switched everything to DataMember stuff in netcore and we didn't have that setting?

Reply:
I think it is because we have our own UmbracoJsonModelBinderConvention now..